src/lsp-server/src/backend.rs

```diff
@@ -91,37 +105,118 @@ pub struct Backend {
     /// prior to save the file.
     pub parse_results: Mutex<HashMap<Url, ParseResult>>,
 
+    /// Contains the latest content for each file.
+    pub content: Mutex<HashMap<Url, Rope>>,
+
     /// Formatting width to use in mz- prettier
     pub formatting_width: Mutex<usize>,
+
+    /// Schema available in the client
+    /// used for completion suggestions.
+    pub schema: Mutex<Option<Schema>>,
+
+    /// Completion suggestion to return
+    /// to the client when requested.
+    pub completions: Mutex<Completions>,
+}
+
+/// Represents a column from an [ObjectType
+#[derive(Debug, Clone, Deserialize)]
+pub struct SchemaObjectColumn {
+    /// Represents the column's name.
+    pub name: String,
+    /// Represents the column's type.
+    #[serde(rename = "type")]
+    pub _type: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+/// Represents each possible object type admissible by the LSP.
+pub enum ObjectType {
+    /// Represents a materialized view.
+    MaterializedView,
+    /// Represents a view.
+    View,
+    /// Represents a source.
+    Source,
+    /// Represents a table.
+    Table,
+    /// Represents a sink.
+    Sink,
+}
+
+impl ToString for ObjectType {
```

Comment:
Generally we would implement `std::fmt::Display` instead, and you get to string for free.